source/bcaa.d

```diff
@@ -481,3 +493,146 @@ unittest {
     Bcaa!(int, int) emptyMap;
     assert(0 !in emptyMap);
 }
+
+/// Verifies that an Struct or Class have the expected API for Allocator
+enum bool isAllocator(T) = __traits(hasMember, T, "make") && __traits(hasMember, T, "dispose");
```

Comment:
Suggestions of how improve this ?

Reply:
Hi Luis, I am kinda busy. But I took a quick look. It looks good. It can be merged. I will review it soon.

Reply:
We can disscuss why we need SimpleAllocator since this one[1] already exists. bcaa uses betterC, and phobos would not be available sometimes?

1: https://dlang.org/phobos/std_experimental_allocator_mallocator.html

Reply:
I closed, because I was like to try first a little experment to see if the Phobos Mallocator counts as @nogc or not. This old post ( https://forum.dlang.org/post/zcajsyoaskdwwcathask@forum.dlang.org ), says that it's inferred as @nogc.